src/types.ts

```diff
@@ -1,30 +1,38 @@
-import { Hover, MarkedString, MarkupContent, Range } from 'vscode-languageserver-types'
+import { Position, Range } from '@sourcegraph/extension-api-types'
 
 export const LOADING: 'loading' = 'loading'
 
-/** LSP proxy error code for unsupported modes */
-export const EMODENOTFOUND = -32000
+/**
+ * Describes the range in the document (usually a token) that the hover is attached to.
+ */
+export interface HoverAttachment {
+    /**
+     * The range to which this hover applies. When missing, it will use the range at the current
+     * position or the current position itself.
+     */
+    range?: Range
+}
 
-/** A hover that is merged from multiple Hover results and normalized. */
-export type HoverMerged = Pick<Hover, Exclude<keyof Hover, 'contents'>> & {
-    /** Also allows MarkupContent[]. */
-    // tslint:disable-next-line deprecation We want to handle MarkedString
-    contents: (MarkupContent | MarkedString)[]
+/**
+ * Reports whether {@link value} is a {@link HoverAttachment} value with a range.
+ */
+export function isHoverAttachmentWithRange(value: any): value is HoverAttachment & { range: Range } {
+    return (
+        value &&
+        value.range &&
+        value.range.start &&
+        typeof value.range.start.line === 'number' &&
+        typeof value.range.start.character === 'number' &&
+        typeof value.range.end.line === 'number' &&
+        typeof value.range.end.character === 'number'
```

Comment:
This removes vscode-languageserver-types so that is no longer available

Reply:
Imo that's a reason to keep it. I did the PR to vscode-languageserver-types that added these and they have test coverage. Why maintain copy and paste them here? Some other part of our app will have that dependency already anyway so this increases the bundle size

Reply:
It just so happens that our Position types line up with theirs, but that might not always be the case or it might be true for other types that are in both `@sourcegraph/extension-api-types` and `vscode-languageserver-types`.

I also worry that people will start importing these in our main codebase or in extensions and get confused.

Anyway, I removed these and inlined them where needed in the last commit.